sky/global_user_state.py

```diff
@@ -359,7 +537,41 @@ def get_clusters() -> List[Dict[str, Any]]:
             'to_down': bool(to_down),
             'owner': owner,
             'metadata': json.loads(metadata),
+            'cluster_hash': cluster_hash,
         }
+
+        records.append(record)
+    return records
+
+
+def get_clusters_from_history() -> List[Dict[str, Any]]:
+    rows = _DB.cursor.execute('SELECT * from cluster_history').fetchall()
+
+    records = []
+
+    # sort by launch time, descending in recency
+    rows.sort(key=lambda row: -_get_cluster_launch_time(row[0]))
+
+    for (
+            cluster_hash,
+            name,
+            num_nodes,
+            _,
+            launched_resources,
+            usage_intervals,
+    ) in rows:
+        # TODO: use namedtuple instead of dict
+
+        record = {
+            'name': name,
+            'launched_at': _get_cluster_launch_time(cluster_hash),
+            'duration': _get_cluster_duration(cluster_hash),
+            'num_nodes': num_nodes,
+            'resources': pickle.loads(launched_resources),
+            'cluster_hash': cluster_hash,
+            'usage_intervals': pickle.loads(usage_intervals),
+        }
+
         records.append(record)
     return records
```

Comment:
How about we sort it here, instead of the previous line at L553?
```suggestion
    records = sorted(records, key=lambda record: record['launched_at'])
    return records
```